tests/src/events-provider.request-scoped.consumer.ts

```diff
@@ -0,0 +1,24 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { OnEvent } from '../../lib';
+import { EVENT_REF } from '../../lib';
+
+@Injectable()
+export class EventsProviderRequestScopedConsumer {
+  constructor(@Inject(EVENT_REF) public eventRef: any) {
+    if (Array.isArray(this.eventRef)) {
+      EventsProviderRequestScopedConsumer.injectedEventMultiPayload =
+        this.eventRef;
+    } else {
+      EventsProviderRequestScopedConsumer.injectedEventPayload = this.eventRef;
+    }
+  }
+
+  public static injectedEventPayload = {};
```

Comment:
The reason why I created this static property is to make it easy to check the event payload was injected in the tests. Otherwise we would need a specific `contextId`